From 5a0ebbf55515a928acf9926d3fcd789115e347d3 Mon Sep 17 00:00:00 2001
From: Lang Daniel <d.lang@abatec.at>
Date: Mon, 17 Jan 2022 13:57:55 +0100
Subject: [PATCH] cmake: improve include paths

Adding the dbus-cxx subfolders as an include path might override system
headers (signal.h for gcc 10).
By removing the subfolders, includes can happen via #include "..." or
via #include <dbus-cxx/...>.

Fetch from: https://github.com/dbus-cxx/dbus-cxx/pull/86
Upstream-Status: Accepted
Signed-off-by: Daniel Lang <d.lang@abatec.at>

---
 CMakeLists.txt                | 4 +---
 dbus-cxx/matchrule.h          | 2 +-
 dbus-cxx/sasl.h               | 2 +-
 dbus-cxx/sendmsgtransport.cpp | 2 +-
 dbus-cxx/simpletransport.cpp  | 2 +-
 5 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b6d28c..e76cd61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -207,9 +207,7 @@ set( DBUS_CXX_HEADERS
 
 set( DBUS_CXX_INCLUDE_DIRECTORIES 
     ${PROJECT_SOURCE_DIR} 
-    ${PROJECT_SOURCE_DIR}/dbus-cxx
-    ${PROJECT_BINARY_DIR}
-    ${PROJECT_BINARY_DIR}/dbus-cxx )
+    ${PROJECT_BINARY_DIR} )
 include_directories( ${DBUS_CXX_INCLUDE_DIRECTORIES} 
     ${dbus_INCLUDE_DIRS} 
     ${sigc_INCLUDE_DIRS} )
diff --git a/dbus-cxx/matchrule.h b/dbus-cxx/matchrule.h
index 69fdab7..18c9e81 100644
--- a/dbus-cxx/matchrule.h
+++ b/dbus-cxx/matchrule.h
@@ -20,7 +20,7 @@
 #define DBUSCXX_MATCH_RULE_H
 
 #include <memory>
-#include "dbus-cxx-config.h"
+#include <dbus-cxx/dbus-cxx-config.h>
 
 namespace DBus {
 
diff --git a/dbus-cxx/sasl.h b/dbus-cxx/sasl.h
index d7429a3..44edd6e 100644
--- a/dbus-cxx/sasl.h
+++ b/dbus-cxx/sasl.h
@@ -19,7 +19,7 @@
 #ifndef DBUSCXX_SASL_H
 #define DBUSCXX_SASL_H
 
-#include <dbus-cxx-config.h>
+#include <dbus-cxx/dbus-cxx-config.h>
 
 #include <memory>
 #include <stdint.h>
diff --git a/dbus-cxx/sendmsgtransport.cpp b/dbus-cxx/sendmsgtransport.cpp
index ae053ff..a10ae66 100644
--- a/dbus-cxx/sendmsgtransport.cpp
+++ b/dbus-cxx/sendmsgtransport.cpp
@@ -21,8 +21,8 @@
 #include "dbus-cxx-private.h"
 #include "utility.h"
 #include "validator.h"
+#include "message.h"
 
-#include <message.h>
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
diff --git a/dbus-cxx/simpletransport.cpp b/dbus-cxx/simpletransport.cpp
index f291d9f..fed4364 100644
--- a/dbus-cxx/simpletransport.cpp
+++ b/dbus-cxx/simpletransport.cpp
@@ -18,7 +18,7 @@
  ***************************************************************************/
 #include "simpletransport.h"
 
-#include <dbus-cxx-private.h>
+#include "dbus-cxx-private.h"
 #include "demarshaling.h"
 #include "message.h"
 #include "utility.h"

